services/horizon/internal/ingest/processors/effects_processor.go

```diff
@@ -50,14 +53,29 @@ func (p *EffectProcessor) ProcessTransaction(
 		return nil
 	}
 
-	for opi, op := range transaction.Envelope.Operations() {
+	elidedTransaction := transaction
+
+	if p.skipSoroban &&
+		elidedTransaction.UnsafeMeta.V == 3 &&
+		elidedTransaction.UnsafeMeta.MustV3().SorobanMeta != nil {
```

Comment:
nit: just accessing the struct will be slightly faster since you did the `V == 3` check already
```suggestion
		elidedTransaction.UnsafeMeta.V3.SorobanMeta != nil {
```

Reply:
ok, i like the encapsulated getter, but there's so much usage of the struct way, changed it.